plugins/git/service/src/gitservice.cpp

```diff
@@ -299,14 +293,16 @@ void GitServiceHandler::getReferenceTopObject(
   int error = git_reference_name_to_id(&oid, repo.get(), branchName_.c_str());
 
   if (error)
-    LOG(error) << "Lookup a reference by name failed: " << error;
+    LOG(warning) << "Lookup a reference by name failed: " << error;
 
   return_.oid = gitOidToString(&oid);
 
   ObjectPtr object = createObject(repo.get(), oid);
 
   if (object)
     return_.type =
+      // TODO: Is this a correct mapping between decltype(return_.type) and
+      // git_object_type()?
```

Comment:
I checked it, this is a correct mapping between these types, but we have to add two more extra types for the `GitObjectType` in `git.thrift` file:
```thrift
GIT_OBJ_ANY = -2,  /**< Object can be any of the following */
GIT_OBJ_BAD = -1,  /**< Object is invalid. */
```